test/unit/transformer-oracle.spec.ts

```diff
@@ -202,6 +202,102 @@ describe('TransformerOracle', () => {
     }
   });
 
+  describe('getRecursiveMappingForPair', () => {
+    const UNDERLYING_UNDERLYING_TOKEN_A = '0x0000000000000000000000000000000000000005';
+    const UNDERLYING_UNDERLYING_TOKEN_B = '0x0000000000000000000000000000000000000006';
+
+    getRecursiveMappingForPairTest({
+      when: 'both tokens have underlying',
+      underlyings: {
+        [TOKEN_A]: UNDERLYING_TOKEN_A,
+        [TOKEN_B]: UNDERLYING_TOKEN_B,
+      },
+      expectedA: UNDERLYING_TOKEN_A,
+      expectedB: UNDERLYING_TOKEN_B,
+    });
+
+    getRecursiveMappingForPairTest({
+      when: 'only tokenA has underlying',
+      underlyings: {
+        [TOKEN_A]: UNDERLYING_TOKEN_A,
+      },
+      expectedA: UNDERLYING_TOKEN_A,
+      expectedB: TOKEN_B,
+    });
+
+    getRecursiveMappingForPairTest({
+      when: 'only tokenB has underlying',
+      underlyings: {
+        [TOKEN_B]: UNDERLYING_TOKEN_B,
+      },
+      expectedA: TOKEN_A,
+      expectedB: UNDERLYING_TOKEN_B,
+    });
+
+    getRecursiveMappingForPairTest({
+      when: 'neither of the tokens have underlying',
+      expectedA: TOKEN_A,
+      expectedB: TOKEN_B,
+    });
+
+    getRecursiveMappingForPairTest({
+      when: 'both tokens have recursive mappings',
+      underlyings: {
+        [TOKEN_A]: UNDERLYING_TOKEN_A,
+        [TOKEN_B]: UNDERLYING_TOKEN_B,
+        [UNDERLYING_TOKEN_A]: UNDERLYING_UNDERLYING_TOKEN_A,
+        [UNDERLYING_TOKEN_B]: UNDERLYING_UNDERLYING_TOKEN_B,
+      },
+      expectedA: UNDERLYING_UNDERLYING_TOKEN_A,
+      expectedB: UNDERLYING_UNDERLYING_TOKEN_B,
+    });
+
+    getRecursiveMappingForPairTest({
+      when: 'only tokenA has a recursive mapping',
+      underlyings: {
+        [TOKEN_A]: UNDERLYING_TOKEN_A,
+        [TOKEN_B]: UNDERLYING_TOKEN_B,
+        [UNDERLYING_TOKEN_A]: UNDERLYING_UNDERLYING_TOKEN_A,
+      },
+      expectedA: UNDERLYING_UNDERLYING_TOKEN_A,
+      expectedB: UNDERLYING_TOKEN_B,
+    });
```

Comment:
Since you've already parameterized this... could we add `only tokenB has recursive mapping`? I know it basically tests the same thing but... It's a cheap test and we complete all branches coverages 😬